Framework/include/QualityControl/CheckRunner.h

```diff
@@ -198,6 +206,10 @@ class CheckRunner : public framework::Task
   // Checks cache
   std::map<std::string, std::shared_ptr<MonitorObject>> mMonitorObjects;
 
+  // Service discovery
+  std::shared_ptr<ServiceDiscovery> mServiceDiscovery;
```

Comment:
I would have liked a unique_ptr but a `Task` cannot have a unique_ptr for some reasons.

Reply:
Do you know why? I'm just curious.

Reply:
no, and I am trying to reproduce the problem in a minimal setup to report it.

Reply:
I think it is because `Task` is transformed into `DataProcessorSpec` and the latter is copied somewhere in DPL.

Reply:
It is a rabbit hole which starts with `CheckRunner` not having a default copy constructor due to `mServiceDiscovery` not having one. It cannot have one because of `curlHandle` not having a move constructor somehow. 

It is certainly fixable but I am not sure that it is worth the trouble. Would you be fine with having the shared_ptr instead ?

Reply:
Fine with me. I don't think there is any risk of concurrent access here.